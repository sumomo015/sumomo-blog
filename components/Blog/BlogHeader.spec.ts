// @vitest-environment nuxt
import { describe, it, expect } from 'vitest'
import { mountSuspended } from '@nuxt/test-utils/runtime'
import type { BlogCollectionItem } from '@nuxt/content'

import { BlogHeader } from '#components'

describe('BlogHeader', () => {
  const article: BlogCollectionItem = {
    id: 'test-article',
    title: 'Test Article',
    description: 'This is a test article',
    body: {
      type: 'root',
      children: [],
    },
    path: '/blog/test-article',
    seo: {},
    extension: '.md',
    meta: {},
    stem: 'test-article',
    datePublished: '2023-01-01',
    dateModified: '2023-01-02',
  }

  it('正常にレンダリングされること', async () => {
    const wrapper = await mountSuspended(BlogHeader, {
      props: { article },
    })
    expect(wrapper.exists()).toBe(true)
  })

  it('タイトルが表示されること', async () => {
    const wrapper = await mountSuspended(BlogHeader, {
      props: { article },
    })
    expect(wrapper.find('h1').text()).toBe('Test Article')
  })

  it('投稿日が表示されること', async () => {
    const wrapper = await mountSuspended(BlogHeader, {
      props: { article },
    })
    const timeElement = wrapper.find('time')
    expect(timeElement.exists()).toBe(true)
    expect(timeElement.text()).toContain('投稿日 2023-01-01')
  })

  it('最終更新日が表示されること', async () => {
    const wrapper = await mountSuspended(BlogHeader, {
      props: { article },
    })
    const timeElements = wrapper.findAll('time')
    expect(timeElements.length).toBe(2)
    expect(timeElements[1]?.text()).toContain('最終更新日 2023-01-02')
  })

  it('最終更新日がない場合、表示されないこと', async () => {
    const wrapper = await mountSuspended(BlogHeader, {
      props: { article: { ...article, dateModified: undefined } },
    })
    const timeElements = wrapper.findAll('time')
    expect(timeElements.length).toBe(1)
    expect(timeElements[0]?.text()).not.toContain('最終更新日')
  })

  it('スナップショット', async () => {
    const wrapper = await mountSuspended(BlogHeader, {
      props: { article },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
