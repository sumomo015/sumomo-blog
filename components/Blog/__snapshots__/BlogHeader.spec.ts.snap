// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BlogHeader > スナップショット 1`] = `
"<div>
  <div class="mb-10"><a href="/blog" class="base-color-link">← 記事一覧へ戻る</a></div>
  <h1 class="mb-4 text-4xl font-bold base-color-header">Test Article</h1>
  <p class="base-color-secondary"><time datetime="2023-01-01"> 投稿日 2023-01-01</time><span> ・</span><time datetime="2023-01-02"> 最終更新日 2023-01-02</time></p>
</div>"
`;
